File: KB boot/week7/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for (int s : scoville) {
            pq.add(s);
        }

        while (pq.size() >= 2 && pq.peek() < K) {
            int a = pq.poll();
            int b = pq.poll();
            int mixed = a + (b * 2);
            pq.add(mixed);
            answer++;
        }

        if (pq.peek() < K) return -1;

        return answer;
    }
}

File: Programmers/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> minHeap;
    
    // 1. 최소 힙 만들기
    for (int i = 0; i < scoville.size(); i++) {
        minHeap.push(scoville[i]);
    }
    
    // 2. 연산
    while (minHeap.top() < K) {
        if (minHeap.size() <= 1) {
            return -1;
        }

        int first = minHeap.top();
        minHeap.pop();
        int second = minHeap.top();
        minHeap.pop();
        
        int mixed = first + (second * 2);
        minHeap.push(mixed);
        answer++;
    }
    
    return answer;
}

